docs/proposals/network-policy.md

```diff
@@ -0,0 +1,259 @@
+
+When `net.beta.kubernetes.io/network-isolation=true` is set on a namespace:
+- Pods in that Namespace will not be accessible from any other source, even sources within that namespace, unless explicitly allowed by a NetworkPolicy object.
+- Pods in that Namespace will be able to access any other destination.
+
+When `net.beta.kubernetes.io/network-isolation=false` is set on a namespace:
+- Pods in that Namespace will be accessible from any other source.
+- Pods in that Namespace will be able to access any other destination.
+
+This field will default to `false` if not specified.
+
+### NetworkPolicy Go Definition 
+
+Once the namespace is isolated, a mechanism to selectively allow traffic into the namespace and between pods within
+the namespace is required.  That is accomplished through ingress rules on `NetworkPolicy`
+objects (of which there can be multiple in a single namespace).  Pods selected by
+one or more NetworkPolicy objects should allow any incoming connections that match any
+ingress rule on those NetworkPolicy objects, per the network plugin’s capabilities.
+
+If `net.beta.kubernetes.io/network-isolation=false` on a namespace, then all traffic is allowed to / from pods in that namespace whether or not any NetworkPolicy objects have been created in that namespace. 
+
+```go
+type NetworkPolicy struct {
+  TypeMeta
+  ObjectMeta
+
+  // Specification of the desired behavior for this NetworkPolicy.
+  Spec NetworkPolicySpec 
+}
+
+type NetworkPolicySpec struct {
+  // Selects the pods to which this NetworkPolicy object
+  // applies.  The array of NetworkPolicyIngressRules below 
+  // is applied to any pods selected by this field.
+  // Multiple NetworkPolicy objects can select the same set of pods.  In this case,
+  // the NetworkPolicyRules for each are combined additively.
+  PodSelector *unversioned.LabelSelector `json:"podSelector,omitempty"`
+
+  // List of ingress rules to be applied to the selected pods.
+  // Traffic is allowed to a pod if network-isolation=false, 
+  // if the traffic source is the pod's host (for kubelet health checks), 
+  // or if network-isolation=true and the traffic matches at least 
+  // one NetworkPolicyIngressRule across all of the NetworkPolicy 
+  // objects whose podSelector matches the pod.  If this field is 
+  // omitted, this NetworkPolicy has no effect on selected pods.
+  Ingress []NetworkPolicyIngressRule `json:"ingress,omitempty"`
+
+type NetworkPolicyIngressRule struct {
+  // List of ports which should be made accessible on the pods selected by PodSelector.  
+  // Each item in this list is combined using a logical OR.  If this field
+  // is omitted, then traffic should not be restricted based on port.  If this 
+  // field is present, traffic must match at least one NetworkPolicyPort in the list
+  // or else it will be dropped.
+  Ports []NetworkPolicyPort `json:"ports,omitempty"`
```

Comment:
OK, this is where I should have asked.  Can this golang representation distinguish between `Ports` missing and `Ports` holding an empty list?